src/pages/CreateNimiPage/CreateNimiPage.tsx

```diff
@@ -10,15 +9,15 @@ import { SUPPORTED_CHAIN_IDS } from '../../constants';
 
 export function CreateNimiPage() {
   const { account, provider, chainId } = useWeb3React();
-  const { ensName } = useParams();
-  const nodeHash = ensNameHash(ensName as string);
+
+  const { state }: any = useLocation();
```

Comment:
I will accept this, but this is NOT ideal. Should be either a Context or redux to handle the load times.